### DISCUSSION TITLE
Compaction / truncation of long-running logs

### DISCUSSION THREAD

**Discussion by psvensson** [royaltm/node-zmq-raft#20] [General]
Hi, I have been wondering how node-zmq-raft handles the growing of logs over time. 

If a raft cluster is up a long time and have a lot of activity, wouldn't the logs eventually overflow the disk, or if not the addition of a new node might be very slow? Perhaps this is handled somewhere, but I haven't found anything (to me) obvious while looking around the code.

I would think that any method to handle this would be necessity have to involve the state machine.

BR,
PS

**Comment by royaltm** [comment]
https://github.com/royaltm/node-zmq-raft/blob/master/RAFT.md#log-compaction

**Comment by psvensson** [comment]
Thank you! 

I did not understand it the first time I read it some months ago, but now I do. 
This is really nice!
